src/node_contractor.rs

```diff
@@ -56,15 +56,13 @@ pub fn handle_shortcuts<F>(
 ) where
     F: FnMut(&mut PreparationGraph, Shortcut),
 {
-    dijkstra.avoid_node(node);
     for i in 0..graph.in_edges[node].len() {
         for j in 0..graph.out_edges[node].len() {
```

Comment:
Note that nothing is sorted here. I think what we need to do is run these loops to find all pairs of incoming/outgoing edges and then sort these pairs by `weight=weight_in_edge+weight_out_edge`. Once they are sorted we need to loop through these pairs and add the required shortcuts. Well, at least this is what I had in mind, maybe I still do not understand what you meant to do.

Reply:
one question:
at every time while contracting a shortcut between 'a' and 'b' is unique and having the minimum weight. is that right?
from your README:
> in case the graph has duplicate edges (multiple edges from node A to node B) only the edge with the lowest weight will be considered

is this done after contraction? or is this ensured the whole time?

Reply:
Yes we exclude duplicate edges (keeping only the one with smallest weight) when creating the input graph. During contraction its done in a way that no duplicate shortcuts can be created (in case a duplicate is about to be created the weight of the existing one is reduced instead).

Reply:
okay so the maximum number of edges (including shortcuts) for all nodes from node a to node b is exactly one; even when contracting. right?
if this is the case my code is exactly meant to be like this.
Can you elaborate why you want to sort the edge-pairs?

you have to iterate over all of them eventually.
in my implementation I am using a `BTree` for de-duplication, so no "in-/out_node" appears two times.
[this part is for deduplication](https://github.com/Stunkymonkey/osm_ch/blob/master/pre/src/contraction.rs#L24-L46)
[this part is the same like in your code](https://github.com/Stunkymonkey/osm_ch/blob/master/pre/src/contraction.rs#L49-L74)

Reply:
> okay so the maximum number of edges (including shortcuts) for all nodes from node a to node b is exactly one; even when contracting. right?

yes that should be the case

Reply:
> Can you elaborate why you want to sort the edge-pairs?

I tried explaining this for the example graph (see comment below)

Reply:
sorry i missed that the last time